core/src/components/datetime/datetime.tsx

```diff
@@ -308,57 +308,78 @@ export class Datetime implements ComponentInterface {
    */
   @Prop() firstDayOfWeek = 0;
 
+  /**
+   * If `true`, multiple dates can be selected at once. Only
+   * applies to `presentation="date"` and `preferWheel="false"`.
+   */
+  @Prop() multiple = false;
+
   /**
    * The value of the datetime as a valid ISO 8601 datetime string.
+   * Will be an array of strings if `multiple="true"`.
    */
-  @Prop({ mutable: true }) value?: string | null;
+  @Prop({ mutable: true }) value?: string | string[] | null;
 
   /**
    * Update the datetime value when the value changes
    */
   @Watch('value')
   protected valueChanged() {
+    const { value, minParts, maxParts, workingParts, multiple } = this;
+
     if (this.hasValue()) {
+      if (!multiple && Array.isArray(value)) {
+        this.value = value[0];
```

Comment:
Is this a safe guard for if someone passes an array to `value` when `multiple` is not true? 

This causes a weird state inconsistency between the developer's state (`["someString"]`) and the value of the datetime element `"someString"`). This also does not appear to emit `ionChange`, so the developer is unable to get their state in-sync with this.

If this is a safe guard, I think we should throw a warning instead.

Reply:
Makes sense, thanks for the explanation 👍